hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyServerRpcConnection.java

```diff
@@ -101,9 +106,7 @@ void process(ByteBuff buf) throws IOException, InterruptedException {
 
   @Override
   public synchronized void close() {
-    disposeSasl();
     channel.close();
-    callCleanupIfNeeded();
```

Comment:
Hmm not calling this is also a problem, but that's fine we can track that later.

Reply:
We register close hook, so close the channel will trigger the close hook and call these two methods.

Reply:
No I didn't mean call to `disposeSasl()` and `callCleanupIfNeeded()`. I meant this whole `NettyServerRpcConnection#close` not getting called is a problem.

Having the channel close hook to dispose sasl is the right thing to do but I think we can simplify this a bit and maybe remove close() from NettyServerRpcConnection or do some better refactoring so that first time readers don't get confused and add more closable stuffs in `NettyServerRpcConnection#close`.